pkg/s3select/select.go

```diff
@@ -147,24 +150,21 @@ func (reader *Input) ParseSelect(sqlInput string) ([]string, string, int64, inte
 		if stmt.OrderBy != nil {
 			return nil, "", 0, nil, nil, nil, ErrParseUnsupportedToken
 		}
-		if err := reader.parseErrs(columnNames, whereClause, alias, myFuncs); err != nil {
+		if err := parseErrs(columnNames, whereClause, alias, myFuncs, inputType); err != nil {
 			return nil, "", 0, nil, nil, nil, err
 		}
 		return columnNames, alias, limit, whereClause, functionNames, myFuncs, nil
 	}
 	return nil, "", 0, nil, nil, nil, nil
 }
 
-// This is the main function, It goes row by row and for records which validate
-// the where clause it currently prints the appropriate row given the requested
-// columns.
-func (reader *Input) processSelectReq(reqColNames []string, alias string, whereClause interface{}, limitOfRecords int64, functionNames []string, myRow chan *Row, myFunc *SelectFuncs) {
+func processSelectReq(reqColNames []string, alias string, whereClause interface{}, limitOfRecords int64, functionNames []string, myRow chan *Row, myFunc *SelectFuncs, inputType SelectQuery) {
```

Comment:
Can you add a comment about this function?

Reply:
Comment added